src/ivan_cherednikov_ex/ex05/bounded_sim.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Ivan Cherednikov'
+__email__ = 'ivch@nmbu.no'
+
+
+from walker_sim import Walker, Simulation
+
+
+class BoundedWalker(Walker):
+    def __init__(self, start, home, left_limit, right_limit):
+        """
+        Initialise the walker
+
+        Arguments
+        ---------
+        start : int
+            The walker's initial position
+        home : int
+            The walk ends when the walker reaches home
+        left_limit : int
+            The left boundary of walker movement
+        right_limit : int
+            The right boundary  of walker movement
+        """
+        super().__init__(start, home)
+        self.left_limit = left_limit
+        self.right_limit = right_limit
+        self.x = start
+
+    def valid_position(self):
+        super().move()
+        if self.x < self.left_limit:
+            self.x = self.left_limit
+            self.step -= 1
+        if self.x > self.right_limit:
+            self.step -= 1
+            self.x = self.right_limit
+
+
+class BoundedSimulation(Simulation):
+    def __init__(self, start, home, seed, left_limit, right_limit):
+        """
+        Initialise the simulation
+
+        Arguments
+        ---------
+        start : int
+             The walker's initial position
+        home : int
+            The walk ends when the walker reaches home
+        seed : int
+            Random generator seed
+        left_limit : int
+            The left boundary of walker movement
+        right_limit : int
+            The right boundary  of walker movement
+        """
+        super().__init__(start, home, seed)
+        self.right_limit = right_limit
+        self.left_limit = left_limit
+
+    def bounded_single_walk(self):
```

Comment:
Instead of defining a new function, which will never be used, you should override `single_walk` here with your new code, then `run_simulation` will use your newly overridden `single_walk`